File: Boj.단계별/9_기본수학2/javascript/2581.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');
const m = Number(input[0]);
const n = Number(input[1]);
const primeNumber = new Array(n+1);
let j, result = 0 , min_prime = 0;
primeNumber[0] = primeNumber[1] = true;
for (let i = 2 ; i<parseInt(Math.sqrt(n))+1 ; i++){
    j = 2;
    while (i*j <= n){
        primeNumber[i*j] = true;
        j++;
    }
}
for (let i = m ; i<= n ; i++){
    if (primeNumber[i]) continue;
    if (!min_prime) min_prime = i;
    result += i
}
if (result === 0) console.log(-1)
else{
    console.log(result);
    console.log(min_prime);
}

File: Boj.단계별/9_기본수학2/python/2581.py
m = int(input())
n = int(input())
result = 0
min_prime = False;
prime = [False for _ in range(n+1)]
prime[0] = prime[1] = True
for i in range(2,n):
    j = 2
    while i*j <= n:
        prime[i*j] = True
        j+=1
for i in range(m,n+1):
    if prime[i]: continue
    if not min_prime : min_prime = i
    result += i
if result == 0 : print(-1)
else:
    print(result)
    print(min_prime)